portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -29,46 +32,34 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String commentText = getCommentText(request);
-    String commentAuthor = getCommentAuthor(request);
+    // Read from request
+    StringBuffer jb = new StringBuffer();
+    String line = null;
+    BufferedReader reader = request.getReader();
+    while ((line = reader.readLine()) != null) {
+      jb.append(line);
+    }
+
+    // Convert request payload to a json
+    String json = jb.toString();
+    JsonObject convertedObject = new Gson().fromJson(json, JsonObject.class);
+    
+    String commentText = getAttribute(convertedObject, "text");
+    String commentAuthor = getAttribute(convertedObject, "author");
     long commentTime = System.currentTimeMillis();
 
+    // Add comment to Datastore
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("text", commentText);
     commentEntity.setProperty("author", commentAuthor);
     commentEntity.setProperty("time", commentTime);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
-
-    String referer = request.getHeader("Referer");
-
-    response.sendRedirect(referer);
   }
 
-  /** Returns the text entered by the user. */
-  private String getCommentText(HttpServletRequest request) {
-    // Get the input from the form.
-    String text = request.getParameter("comment-text");
-    
-    // Account for a blank response.
-    if (text == null) {
-      return "";
-    }
-
-    return text;
-  }
-
-  /** Returns the author entered by the user, or "Anonymous" if left blank. */
-  private String getCommentAuthor(HttpServletRequest request) {
-    // Get the input from the form.
-    String author = request.getParameter("comment-author");
-    
-    // Account for a blank response.
-    if (author == null || author.equals("")) {
-      return "Anonymous";
-    }
-
-    return author;
+  // Retrieves and removes quotes around the specified attribute of obj
+  private String getAttribute(JsonObject obj, String attr) {
+    return obj.get(attr).toString().replaceAll("^\"|\"$", "");
```

Comment:
Did you try `getString()` instead of `get()` by any chance? I'm wondering if it might automatically remove the quotes for you.

Reply:
That's what I initially tried, but the [JsonObject class](https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5/com/google/gson/JsonObject.html) in gson dooesn't a method that does what I want. `get()` returns a [JsonElement](https://www.javadoc.io/static/com.google.code.gson/gson/2.8.5/com/google/gson/JsonElement.html) which can be converted to a string, but from the documentation, elements are literally translated, including quotes.